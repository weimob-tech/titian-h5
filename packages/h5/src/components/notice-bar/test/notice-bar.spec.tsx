import { newSpecPage } from '@stencil/core/testing';
import { h } from '@stencil/core';
import { TiNoticeBar } from '../index';
// import { TiPopup } from '../../popup';
// import { TestContainer } from '../../common/test/container';

describe('渲染 ti-notice-bar', () => {
  it('基本用法', async () => {
    const page = await newSpecPage({
      components: [TiNoticeBar],
      template: () => <ti-notice-bar content="内容文字内容文字内容文字内容文字内容文字内容文字内容文字内容文字" />,
    });
    page.root.setAttribute('variant', 'vertical');
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });
  // it('纵向滚动', async () => {
  //   const page = await newSpecPage({
  //     components: [TiNoticeBar],
  //     template: () => <ti-notice-bar content={['内容1', '内容2']} variant="vertical" />,
  //   });
  //   expect(page.root).toMatchSnapshot();
  // });
  // it('滚动速度', async () => {
  //   const page = await newSpecPage({
  //     components: [TiNoticeBar],
  //     template: () => (
  //       <ti-notice-bar content="内容文字内容文字内容文字内容文字内容文字内容文字内容文字内容文字" speed={100} />
  //     ),
  //   });
  //   expect(page.root).toMatchSnapshot();
  // });
  // it('内容超出后显示方式', async () => {
  //   const page = await newSpecPage({
  //     components: [TiNoticeBar, TestContainer],
  //     template: () => (
  //       <test-container>
  //         <ti-notice-bar content="内容文字内容文字内容文字内容文字内容文字内容文字内容文字内容文字" text-mode="auto" />
  //         <ti-notice-bar content="内容文字内容文字内容文字内容文字内容文字内容文字内容文字内容文字" text-mode="wrap" />
  //         <ti-notice-bar
  //           content="内容文字内容文字内容文字内容文字内容文字内容文字内容文字内容文字"
  //           text-mode="ellipsis"
  //         />
  //         <ti-notice-bar content="内容文字内容文字内容文字内容文字内容文字内容文字内容文字内容文字" text-mode="clip" />
  //       </test-container>
  //     ),
  //   });
  //   expect(page.root).toMatchSnapshot();
  // });
  // it('动态改变variant', async () => {
  //   const page = await newSpecPage({
  //     components: [TiNoticeBar],
  //     template: () => <ti-notice-bar content="内容文字内容文字内容文字内容文字内容文字内容文字内容文字内容文字" />,
  //   });
  //   page.root.setAttribute('variant', 'vertical');
  //   await page.waitForChanges();
  //   expect(page.root).toMatchSnapshot();
  // });

  // it('动态改变scrollable', async () => {
  //   const page = await newSpecPage({
  //     components: [TiNoticeBar],
  //     template: () => <ti-notice-bar content="内容文字内容文字内容文字内容文字内容文字内容文字内容文字内容文字" />,
  //   });
  //   page.root.setAttribute('scrollable', 'false');
  //   page.root.setAttribute('color', 'orange');
  //   await page.waitForChanges();
  //   expect(page.root).toMatchSnapshot();
  // });

  // it('在popup中使用', async () => {
  //   const page = await newSpecPage({
  //     components: [TiNoticeBar, TiPopup],
  //     template: () => (
  //       <ti-popup>
  //         <ti-notice-bar content="内容文字内容文字内容文字内容文字内容文字内容文字内容文字内容文字" />,
  //       </ti-popup>
  //     ),
  //   });
  //   page.root.setAttribute('visible', 'true');
  //   await page.waitForChanges();
  //   expect(page.root).toMatchSnapshot();
  // });
});
