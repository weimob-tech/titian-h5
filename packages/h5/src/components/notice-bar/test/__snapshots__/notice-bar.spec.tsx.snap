// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`渲染 ti-notice-bar 基本用法 1`] = `
<ti-notice-bar variant="vertical">
  <mock:shadow-root>
    <div class="titian-notice-bar titian-notice-bar-auto undefined" id="titian-notice-bar">
      <slot name="before"></slot>
      <div class="titian-notice-bar-container">
        <div class="titian-notice-bar-translate">
          <div class="titian-notice-bar-content titian-notice-bar-vertical">
            内容文字内容文字内容文字内容文字内容文字内容文字内容文字内容文字
          </div>
        </div>
      </div>
      <slot name="after"></slot>
    </div>
  </mock:shadow-root>
</ti-notice-bar>
`;
